samples/start-app/python-fastapi/app/tests/usecase/test_callback.py

```diff
@@ -0,0 +1,71 @@
+import unittest
+from unittest.mock import MagicMock, patch
+from fastapi import Request
+from sqlalchemy.orm import Session
+from app.usecase.callback import callback_usecase
+from app.dependencies import get_config
+
+class TestCallbackUsecase(unittest.TestCase):
+
+    def setUp(self):
+        self.request = MagicMock(spec=Request)
+        self.db = MagicMock(spec=Session)
+        self.session = MagicMock()
+        self.session.headers = {'oauth2_state': 'state_value', 'pim_url': 'http://pim-url.com'}
+        self.request.query_params = {'state': 'state_value', 'code': 'auth_code'}
+
+    @patch('app.persistence.tokenRepository.create_token')
+    @patch('requests.post')
+    @patch('secrets.token_hex')
+    @patch('hashlib.sha256')
+    def test_callback_usecase(self, mock_sha256, mock_token_hex, mock_post, mock_create_token):
+        # Arrange
+        mock_post.return_value.json.return_value = {'access_token': 'access_token_value'}
+        mock_sha256.return_value.hexdigest.return_value = 'd9252f527368ffee07ffe7d9c2722d1e244fc94a8e79e938b8e5a815584942cb'
+        mock_token_hex.return_value = '4d713c758a30768e5576cb7b6f3b2dad62e7bdb5fd4a6132b11ca6301824'
+        # Act
+        response = callback_usecase(self.request, self.db, self.session)
+        # Assert
+        mock_post.assert_called_once_with(
+            'http://pim-url.com/connect/apps/v1/oauth2/token',
+            data={
+                'client_id': get_config('CLIENT_ID'),
+                'code_identifier': '4d713c758a30768e5576cb7b6f3b2dad62e7bdb5fd4a6132b11ca6301824', 
+                'code_challenge': 'd9252f527368ffee07ffe7d9c2722d1e244fc94a8e79e938b8e5a815584942cb',
+                'code': 'auth_code',
+                'grant_type': 'authorization_code'
+            },
+            headers={
+                'Content-Type': 'application/x-www-form-urlencoded',
+                'User-Agent': 'AkeneoSampleApp/python-fastapi Version/1.0.0'
```

Comment:
As we don't change User-Agent before building headers, this will fail if someone uses this test in further versions, as well as if a docker-version has been found.

I would recommend using the environment variables in the expected User-Agent.

Reply:
fix